helm/hpcc/templates/_helpers.tpl

```diff
@@ -429,6 +429,34 @@ Pass in dict with root and secretsCategories
 {{- end -}}
 {{- end -}}
 
+{{/*
+Generate Prometheus scrape annotations
+Enables selfdiscovery of metrics service on configured path/port
+Requires metrics.sinks[type=prometheus]
+Pass in metric config
+*/}}
+{{- define "hpcc.addPrometheusScrapeAnnotations" -}}
+{{- if hasKey . "sinks" }}
+ {{ range $sink := .sinks -}}
+  {{ range $sinkattindex, $sinkatts := $sink -}}
+   {{- if and  (eq $sinkattindex "type") }}
```

Comment:
there has to be a straightforward way to test .sinks[type="prometheus"]

Reply:
I think you can simply the "type" test to:
```
{{ range $sink := .sinks -}}
 {{- if eq (get $sink "type") "prometheus" }}
```

get will return "" if missing.